packages/agents/sequencer/src/lib/operations/auctions.ts

```diff
@@ -193,139 +192,187 @@ export const executeAuctions = async (_requestContext: RequestContext) => {
+          const bidsRoundMap = getBidsRoundMap(bids, config.maxPathLength);
+          const availableRoundIds = [...Object.keys(bidsRoundMap)].sort((a, b) => Number(a) - Number(b));
+          if ([...Object.keys(bidsRoundMap)].length < 1) {
+            logger.warn("No rounds available for this transferId", requestContext, methodContext, {
+              bidsRoundMap,
               transferId,
             });
-            // Not enough router bids to form a transfer for this round.
-            // (e.g. for round 3, we need 3 router bids to form a multipath transfer)
+
             return;
           }
 
-          // TODO: Sort by fee amount, selecting the best bid available.
-          // Randomly sort the bids.
-          const randomized = availableBids
-            .map((value) => ({ value, sort: Math.random() }))
-            .sort((a, b) => a.sort - b.sort)
-            .map(({ value }) => value);
-
-          let taskId: string | undefined;
-          // Try every bid until we find one that works.
-          for (const randomBid of randomized) {
-            logger.info("Trying bid", requestContext, methodContext, {
-              transferId,
-              randomBid,
-            });
-            // Sanity: Check if this router has enough funds.
-            const { router } = randomBid;
-            const asset = await getDestinationLocalAsset(
-              transfer.originDomain,
-              transfer.origin.assets.bridged.asset,
-              destination,
-            );
-            // TODO: Should use amount from router's bid.
-            const amount = transfer.origin.assets.bridged.amount;
-
-            let routerLiquidity: BigNumber | undefined = await cache.routers.getLiquidity(router, destination, asset);
-            if (!routerLiquidity) {
-              // Either we haven't cached the liquidity yet, or the value cached has become expired.
-              routerLiquidity = await subgraph.getAssetBalance(destination, router, asset);
-              if (!routerLiquidity.eq(constants.Zero)) {
-                await cache.routers.setLiquidity(router, destination, asset, routerLiquidity);
-              } else {
-                // NOTE: Using WARN level here as this is unexpected behavior... routers who are bidding on a transfer should
-                // have added liquidity for the asset on the corresponding domain.
-                logger.warn("Skipped bid from router; liquidity not found in subgraph", requestContext, methodContext, {
-                  transfer: {
-                    transferId,
-                    asset,
-                    destination,
-                    amount: amount.toString(),
-                  },
-                  assetBalanceId: `${asset.toLowerCase()}-${router.toLowerCase()}`,
-                  routerLiquidity,
+          for (const roundIdx of availableRoundIds) {
+            logger.debug(`Selecting the round ${roundIdx}`, requestContext, methodContext, { availableRoundIds });
```

Comment:
can we log the number of available bids here? like how many we have to choose from, how many combinations maybe?